sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -73,6 +73,22 @@ def raise_with_traceback(exception, *args, **kwargs):
         error.__traceback__ = exc_traceback
         raise error
 
+class ErrorMap(object):
+    """Error Map class. To be used in map_error method, behaves like a dictionary.
+    It returns the error type if it is found in custom_error_map. Or return default_error
+
+    :param dict custom_error_map: User-defined error map
+    :keyword error default_error: Default error type
+    """
+    def __init__(self, custom_error_map=None, **kwargs):
+        self._custom_error_map = custom_error_map
+        self._default_error = kwargs.pop("default_error", None)
+
+    def get(self, key, value=None):  # pylint:disable=unused-argument
```

Comment:
`map_error` doesn't pass `value`, and we don't use it for anything here. Do we need it?